packages/amplify_datastore/ios/Classes/types/model/FlutterSerializedModel.swift

```diff
@@ -155,6 +160,21 @@ struct FlutterSerializedModel: Model, JSONValueHolder {
 
                 result[key] = deserializedValue
             }
+            else if case .date = field?.type,
+                case .some(.string(let deserializedValue)) = values[key] {
+
+                result[key] = String(deserializedValue.dropLast())
```

Comment:
what's the purpose of `dropLast` here? Are we removing the last character off the string value?

Reply:
droplast is to remove the last Z character.  not removing it will cause a parse exception in Dart

Reply:
Can you add documentation here about what is the Date value coming here in iOS vs what is expected in Dart (and comparing to what comes in android)? So we accurately document such discrepancies between platforms.

Reply:
It only supports for certain formats.  The Date object is returned as yyyy-mm-dd but in order to have Z it would need to be in format yyyy-mm-ddThhZ.  Similar issues for Time.  I think we should drop the Z as Android does not return it in their format.